### PR TITLE
Feature/update ember 3.4

### PR DESCRIPTION
**Pull request by tzellman**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
# 1.2.0 (2018-10-15)
* **Updated** ember-cli to 3.4
* **Updated** updated testing to use QUnit and latest Ember testing API

### CODE REVIEW

**Comment by tzellman** [comment]
I haven't searched online yet, but is there a way to test the travis execution locally prior to a push? I'd like to fix this error.

**Comment by adamlc** [comment]
Looking at the error it may be a problem with travis, if the tests pass locally then its probably a configuration or temp issue with travis.

File: testem.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+  framework: 'mocha',
+  test_page: 'tests/index.html?hidepassed&coverage',
+  disable_watching: true,
+  launch_in_ci: ['Chrome'],
+  launch_in_dev: ['Chrome'],
+  browser_args: {
+    Chrome: {
+      mode: 'ci',
+      args: [
+        '--headless',
```
Comment (jelhan) [inline testem.js:11]: Travis is failing cause it's missing `--no-sandbox` flag in CI. This is how it's done in default blueprint:

```
// --no-sandbox is needed when running Chrome inside a container
process.env.CI ? '--no-sandbox' : null,
```

File: testem.js [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
-}
+  framework: 'mocha',
+  test_page: 'tests/index.html?hidepassed&coverage',
+  disable_watching: true,
+  launch_in_ci: ['Chrome'],
+  launch_in_dev: ['Chrome'],
+  browser_args: {
+    Chrome: {
+      mode: 'ci',
+      args: [
```
Comment (jelhan) [inline testem.js:10]: Testem changed how configuration should be formatted some time ago. Basically mode should now be the key on `Chrome` containing `args` array:

```
Chrome: {
  ci: [
    // args
  ]
};
```

Please compare with [ember-new-output for 3.4.x](https://github.com/ember-cli/ember-new-output/blob/v3.4.3/testem.js).

**Comment by jelhan** [comment]
Currently ember-block-slots throws on build if used with a recent version of ember:

> Uncaught Error: Failed to create an instance of \'component:yield-slot\'. Most likely an improperly defined class or an invalid module export.

This is caused by not using module API but `import Ember from 'ember'`.

To resolve this one, `ember-prop-types` has to be upgraded to `^8.0.0`.

File: package.json [hunk 3]
```diff
@@ -2,78 +2,77 @@
...
-    "named yields",
-    "slots",
-    "ember",
-    "addon",
-    "component"
-  ],
-  "dependencies": {
-    "ember-cli-babel": "^5.1.6",
-    "ember-cli-htmlbars": "^1.0.3",
-    "ember-prop-types": ">=2.0.0 <4.0.0",
```
Comment (jelhan) [inline package.json:72]: Isn't this dependency needed anymore? If it is, it has to be updated to `^8.0.0` to support ember >= 3.3.0 as mentioned in their changelog: https://github.com/ciena-blueplanet/ember-prop-types/blob/master/CHANGELOG.md#800-2018-09-20

**Comment by ncoden** [comment]
@tzellman @jelhan @adamlc What's the status of this pull request?

**Comment by jelhan** [comment]
@ncoden I've refactored to not use ember-block-slots anymore. Just left my findings here cause they may be helpful for others.

**Comment by tzellman** [comment]
We can close this PR. I also have refactored to not use ember-block-slots anymore. It was definitely a very useful addon for several past projects!

In case anyone stumbles upon this comment, the methodology I took in late 2018 is described here: https://dockyard.com/blog/2018/11/26/how-to-yield-an-ember-component-in-multiple-places

**Pull request closed without merging**